src/devices/sound/nes_apu.h

```diff
@@ -64,6 +64,8 @@ class nesapu_device : public device_t,
 	// sound stream update overrides
 	virtual void sound_stream_update(sound_stream &stream, std::vector<read_stream_view> const &inputs, std::vector<write_stream_view> &outputs) override;
 
+	virtual void update_lfsr(apu_t::noise_t *chan);
+
```

Comment:
I don’t think it needs to deal with `chan` being null or being indexed as an array – you could use a reference rather than a pointer.

Reply:
Changed.